### PR TITLE
✨ Comp backend: allow multiple amis/bootscripts/pre-pull images

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
This PR makes the autoscaling service to define an AMI, specific boot scripts, pre-pull images per EC2 instance type instead of one for all.

This means the following ENVs are deprecated in **Autoscaling** service:
```EC2_INSTANCES_AMI_ID```
```EC2_INSTANCES_CUSTOM_BOOT_SCRIPTS```
```EC2_INSTANCES_PRE_PULL_IMAGES```
```EC2_INSTANCES_PRE_PULL_IMAGES_CRON_INTERVAL```

and following ENVs are deprecated in **Clusters-keeper** service:
```WORKERS_EC2_INSTANCES_AMI_ID```
```WORKERS_EC2_INSTANCES_AMI_ID```

The ENV ```EC2_INSTANCES_ALLOWED_TYPES``` and ```WORKERS_EC2_INSTANCES_ALLOWED_TYPES``` are now a dictionary and may be created like so
```python
json.dumps(
  {
    "t2.micro": 
      {
        "ami_id": AMISTRING, 
        "custom_boot_scripts": Optional[list[COMMAND]], 
        "pre_pull_images": Optional[list[IMAGETAG]], 
        "pre_pull_images_cron_interval": Optional[timedelta]
     }
  }
)
```

### Bonus
- partially check the syntax of the custom scripts (at least finds weird quoting), partially fixes https://github.com/ITISFoundation/osparc-simcore/issues/3910
- define all the ENVs variables in docker-compose.yml/.env-devel as required by our new best practices (a corresponding PR in osparc-ops-environment will follow) see https://github.com/ITISFoundation/osparc-ops-environments/pull/438
- fixes wrong ENV variable set for clusters-keeper service in docker-compose (see https://github.com/ITISFoundation/osparc-simcore/issues/5029#issuecomment-1812468479)

## Related issue/s
- https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/148
- https://github.com/ITISFoundation/osparc-ops-environments/pull/438

## Dev Checklist
multiple changes in ENVs for **autoscaling** and **clusters-keeper**
documented in:
- https://github.com/ITISFoundation/osparc-ops-environments/pull/438
- https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/148
- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

File: .env-devel [hunk 1]
```diff
@@ -18,6 +18,12 @@ AGENT_VOLUMES_CLEANUP_S3_PROVIDER=MINIO
 
 API_SERVER_DEV_FEATURES_ENABLED=0
 
+AUTOSCALING_DASK=null
+AUTOSCALING_EC2_ACCESS=null
+AUTOSCALING_EC2_INSTANCES=null
+AUTOSCALING_NODES_MONITORING=null
+AUTOSCALING_POLL_INTERVAL=10
```
Comment (YuryHrytsuk) [inline .env-devel:25]: Are these variables already  in `repo.config`?

**Review by mrnicegyu11 — APPROVED** [review]
tldr thx

**Review by GitHK — APPROVED** [review]
👍

File: .env-devel (hunk 1 above)
Comment (sanderegg) [reply .env-devel:25]: no, not yet. but coming soon. don't worry I will not forget since that will make sim4life.io fail

File: services/autoscaling/src/simcore_service_autoscaling/models.py [hunk 2]
```diff
@@ -62,3 +67,84 @@ class Cluster:
 class DaskTask:
     task_id: DaskTaskId
     required_resources: DaskTaskResources
+
+
+class EC2InstanceBootSpecific(BaseModel):
+    ami_id: str
+    custom_boot_scripts: list[str] = Field(
```
Comment (pcrespov) [inline services/autoscaling/src/simcore_service_autoscaling/models.py:74]: MINOR: perhaps `TypeAlias` many of the `str` around for doc? e.g. in this case `CommandStr: Typealias = str`

Comment (sanderegg) [reply services/autoscaling/src/simcore_service_autoscaling/models.py:74]: done

File: services/autoscaling/requirements/_base.in [hunk 3]
```diff
@@ -19,3 +19,4 @@ aiodocker
 dask[distributed]
 fastapi
 packaging
+sh
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/autoscaling/requirements/_base.in:22]: that might be. it is actuallky nicer to use than subprocess.

File: .env-devel (hunk 1 above)
Comment (sanderegg) [reply .env-devel:25]: done now
